server/routerlicious/packages/protocol-base/src/protocol.ts

```diff
@@ -41,11 +44,46 @@ export function isSystemMessage(message: ISequencedDocumentMessage) {
     }
 }
 
+export interface IProtocolHandler {
+    readonly quorum: IQuorum;
+    readonly attributes: IDocumentAttributes;
+
+    setConnectionState(connected: boolean, clientId: string | undefined);
+    snapshot(): IQuorumSnapshot;
+
+    close(): void;
+    processMessage(message: ISequencedDocumentMessage, local: boolean): IProcessMessageResult;
+}
+
+export type ProtocolHandlerBuilder = (
+    attributes: IDocumentAttributes,
+    snapshot: IQuorumSnapshot,
+    sendProposal: (key: string, value: any) => number,
+) => IProtocolHandler;
+
+export const MakeProtocolHandler: ProtocolHandlerBuilder = (
```

Comment:
I'm not sure I see much value in having MakeProtocolHandler, given that we expect to have different implementations, and class itself has to be exported.